pyquil/pyqvm.py

```diff
@@ -154,6 +154,10 @@ def do_post_gate_noise(self, noise_type: str, noise_prob: float, qubits: List[in
         """
 
 
+# NOTE: both quilc-QVM and qcs-QPU QAMs have asynchronous
+# methods `execute`, `get_result`, and `run`.
+# In order to maintain a common interface these methods do also,
+# despite the fact they are synchronous.
```

Comment:
Should we have a `QAM` and `AsyncQAM`? This would make `get_qc` more complicated, or potentially require splitting it into two functions.

Also note that https://github.com/rigetti/pyquil/issues/1489 may make more operations synchronous.